front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/ServiceAccountsController.groovy

```diff
@@ -34,18 +39,39 @@ public class ServiceAccountsController {
   @Autowired
   ServiceAccountDAO serviceAccountDAO;
 
+  @Autowired(required = false)
+  FiatService fiatService
+
   @RequestMapping(method = RequestMethod.GET)
   Set<ServiceAccount> getAllServiceAccounts() {
     serviceAccountDAO.all();
   }
 
   @RequestMapping(method = RequestMethod.POST)
   ServiceAccount createServiceAccount(@RequestBody ServiceAccount serviceAccount) {
-    return serviceAccountDAO.create(serviceAccount.id, serviceAccount)
+    def acct = serviceAccountDAO.create(serviceAccount.id, serviceAccount)
+    syncUsers(acct)
+    return acct
   }
 
   @RequestMapping(method = RequestMethod.DELETE, value = "/{serviceAccountId:.+}")
   void deleteServiceAccount(@PathVariable String serviceAccountId) {
+    def acct = serviceAccountDAO.findById(serviceAccountId)
     serviceAccountDAO.delete(serviceAccountId)
+    syncUsers(acct)
+  }
+
+  private void syncUsers(ServiceAccount serviceAccount) {
+    if (!fiatService || !serviceAccount) {
+      return
+    }
+
+    def roles = [StringUtils.substringBefore(serviceAccount.name.replaceFirst('%40', "@"), "@")]
```

Comment:
Is `serviceAccount.name` always url-encoded? `replaceFirst` returns the string whether it replaces a character or not -- might be an issue here.